Tests/SourceKitLSPTests/SwiftPMIntegrationTests.swift

```diff
@@ -359,4 +359,74 @@ final class SwiftPMIntegrationTests: XCTestCase {
       ["Cannot convert value of type 'Int' to specified type 'String'"]
     )
   }
+
+  func testToolPlugin() async throws {
+    let project = try await SwiftPMTestProject(
+      files: [
+        "Plugins/plugin.swift": #"""
+        import PackagePlugin
+        @main struct CodeGeneratorPlugin: BuildToolPlugin {
+          func createBuildCommands(context: PluginContext, target: Target) throws -> [Command] {
+            let genSourcesDir = context.pluginWorkDirectoryURL.appending(path: "GeneratedSources")
+            guard let target = target as? SourceModuleTarget else { return [] }
+            let codeGenerator = try context.tool(named: "CodeGenerator").url
+            let generatedFile = genSourcesDir.appending(path: "\(target.name)-generated.swift")
+            return [.buildCommand(
+              displayName: "Generating code for \(target.name)",
+              executable: codeGenerator,
+              arguments: [
+                generatedFile.path
+              ],
+              inputFiles: [],
+              outputFiles: [generatedFile]
+            )]
+          }
+        }
+        """#,
+
+        "Sources/CodeGenerator/CodeGenerator.swift": #"""
+        import Foundation
+        try "let generated = 1".write(to: URL(fileURLWithPath: CommandLine.arguments[1]), atomically: true, encoding: String.Encoding.utf8)
+        """#,
+
+        "Sources/TestLib/TestLib.swift": #"""
+        func useGenerated() {
+          _ = 2️⃣generated
+        }
+        """#,
+      ],
+      manifest: """
+        // swift-tools-version: 6.0
+        import PackageDescription
+        let package = Package(
+          name: "PluginTest",
+          targets: [
+            .executableTarget(name: "CodeGenerator"),
+            .target(
+              name: "TestLib",
+              plugins: [.plugin(name: "CodeGeneratorPlugin")]
+            ),
+            .plugin(
+              name: "CodeGeneratorPlugin",
+              capability: .buildTool(),
+              dependencies: ["CodeGenerator"]
+            ),
+          ]
+        )
+        """,
+      enableBackgroundIndexing: true
+    )
+
+    let (uri, positions) = try project.openDocument("TestLib.swift")
+
+    try await project.testClient.send(PollIndexRequest())
```

Comment:
This shouldn’t be necessary. We should wait for background indexing to finish (by sending a `PollIndexRequest`) when creating the `SwiftPMTestProject`.

Reply:
Thanks, I believe I copy pasted from another test, but may have changed the test project after.